src/library/arith.lisp

```diff
@@ -485,3 +504,118 @@
+         (Quantization f fl-quo fl-rem ce-quo ce-rem)))))
+
+  (define-instance (Quantizable Double-Float)
+    (define (quantize f)
+      (lisp (Quantization Double-Float) (f)
+        (uiop:nest
+         (cl:multiple-value-bind (fl-quo fl-rem) (cl:floor f))
+         (cl:multiple-value-bind (ce-quo ce-rem) (cl:ceiling f))
+         (Quantization f fl-quo fl-rem ce-quo ce-rem)))))
+
+  (define-instance (Quantizable Fraction)
+    (define (quantize q)
+      (let ((n (numerator q))
+            (d (denominator q)))
+        (lisp (Quantization Fraction) (n d)
+          ;; Not the most efficient... just relying on CL to do the
+          ;; work.
+          (cl:flet ((to-frac (f)
+                      (%Fraction (cl:numerator f) (cl:denominator f))))
+            (cl:let ((f (cl:/ n d)))
+              (uiop:nest
+               (cl:multiple-value-bind (fl-quo fl-rem) (cl:floor f))
+               (cl:multiple-value-bind (ce-quo ce-rem) (cl:ceiling f))
+               (Quantization f
+                             fl-quo (to-frac fl-rem)
+                             ce-quo (to-frac ce-rem)))))))))
+
+  (define (floor x)
+    "Return the greatest integer less than or equal to X."
+    (match (quantize x)
+      ((Quantization _ z _ _ _) z)))
+
+  (define (ceiling x)
+    "Return the least integer greater than or equal to X."
+    (match (quantize x)
+      ((Quantization _ _ _ z _) z)))
+
+  (define (round x)
+    "Return the nearest integer to X, with ties breaking toward positive infinity."
+    (match (quantize x)
+      ((Quantization _ a ar b br)
+       (match (<=> (abs ar) (abs br))
+         ((LT) a)
+         ((GT) b)
+         ((EQ) (max a b))))))
+
+  (declare safe/ ((Dividable :a :b) => (:a -> :a -> (Optional :b))))
+  (define (safe/ x y)
+    "Safely divide X by Y, returning None if Y is zero."
+    (if (== y (fromInt 0))
+        None
+        (Some (/ x y))))
+
+  (declare exact/ (Integer -> Integer -> Fraction))
+  (define (exact/ a b)
+    "Exactly divide two integers and produce a fraction."
+    (/ a b))
+
+  (declare inexact/ (Integer -> Integer -> Double-Float))
+  (define (inexact/ a b)
+    "Compute the quotient of integers A and B as a double-precision float.
+
+Note: This does *not* divide double-float arguments."
+    (/ a b))
+
+  (declare floor/ (Integer -> Integer -> Integer))
+  (define (floor/ a b)
+    "Divide two integers and compute the floor of the quotient."
+    (floor (exact/ a b)))
+
+  (declare ceiling/ (Integer -> Integer -> Integer))
+  (define (ceiling/ a b)
+    "Divide two integers and compute the ceiling of the quotient."
+    (ceiling (exact/ a b)))
+
+  (declare round/ (Integer -> Integer -> Integer))
+  (define (round/ a b)
+    "Divide two integers and round the quotient."
+    (round (exact/ a b)))
+
+  (declare single/ (Single-Float -> Single-Float -> Single-Float))
+  (define (single/ a b)
+    "Compute the quotient of single-precision floats A and B as a single-precision float."
```

Comment:
In other places you deleted argument names from the doc string. Should it happen here as well?

Reply:
yes